### PR TITLE
Add unary mode to TabularDataset

### PR DESCRIPTION
**Pull request by Hakuyume**
This PR adds the following syntax.
```python
view = dataset.slice[:, 'a']  # c.f. dataset.slice[:, ('a',)]
a = view[0]
```

### CODE REVIEW

**Comment by Hakuyume** [comment]
jenkins, test this please

File: chainer/dataset/tabular/_slice.py [hunk 1]
```diff
@@ -32,7 +40,12 @@ def keys(self):
 
     @property
     def mode(self):
-        return self._dataset.mode
+        if self._unary is None:
+            return self._dataset.mode
+        elif self._unary:
+            return None
+        else:
+            return self._dataset.mode or tuple
```
Comment (beam2d) [inline chainer/dataset/tabular/_slice.py]: I'm just curious but is it valid to index the key against unary slice (e.g. `dataset.slice[:, 'key'].slice[:, ('key',)]`)?

File: chainer/dataset/tabular/tabular_dataset.py [hunk 2]
```diff
@@ -23,6 +23,8 @@ class TabularDataset(dataset_mixin.DatasetMixin):
     Since an example can be represented by both tuple and dict (
     :obj:`(a[i], b[i], c[i])` and :obj:`{'a': a[i], 'b': b[i], 'c': c[i]}`),
     this class uses :attr:`mode` to indicate which representation will be used.
+    If there is only one column. an example also can be represented by a value
```
[the remark this answers is not in the document]
Comment (Hakuyume) [reply chainer/dataset/tabular/tabular_dataset.py]: Thank you.

File: chainer/dataset/tabular/_slice.py (hunk 1 above)
Comment (Hakuyume) [reply chainer/dataset/tabular/_slice.py]: I think we don't have such use cases. I just thought supporting that makes the API more simple. Do you prefer raising an exception?

Comment (beam2d) [reply chainer/dataset/tabular/_slice.py]: No strong preference on it. I just wanted to confirm to what extent the behavior is intended. OK to leave it as is.

**Comment by beam2d** [comment]
Jenkins, test this please.

**Comment by Hakuyume** [comment]
jenkins, test this please

**Comment by beam2d** [comment]
LGTM. Jenkins failure is not related.

**Pull request merged**